src/v2/components/FireList.jsx

```diff
@@ -93,21 +93,45 @@ export default function FireList(props) {
   const handleVoteForFire = useCallback((index, vote) => {
     if (index > -1 && index < nFires) {
       const fire = fires[index]
-      const {cameraID, timestamp} = fire
-      const isRealFire = vote === 'yes'
-      const isUndo = vote === 'undo'
-      const endpoint = isUndo ? '/api/undoVoteFire' : '/api/voteFire'
-
-      query.post(endpoint, {cameraID, isRealFire, timestamp}).then(() => {
-        if (isUndo) {
-          delete fire.voted
-        } else {
-          fire.voted = isRealFire
-        }
-
-        updateFires(indexOfOldFires > -1)
-      }) // TODO: Implement error handling. Allow errors to go uncaught for now
-         // asy they will be logged to the Console.
+      const promises = [fire]
+        .concat(Object.keys(fire._anglesByKey).map((k) => firesByKey[k]))
+        .reduce((a, f) => {
+          const {cameraID, timestamp} = f
+
+          // Vote only for `fire` and overlapping angles older than `fire`.
+          if (timestamp > fire.timestamp) {
+            return a
+          }
+
+          const hasVote = f.voted != null
+          const isUndo = vote === 'undo'
+
+          // Don’t try to undo nonexistant votes.
+          if (isUndo && !hasVote) {
+            return a
+          }
+
+          const isRealFire = vote === 'yes'
+          const endpoint = isUndo ? '/api/undoVoteFire' : '/api/voteFire'
+
+          const p =
+            query.post(endpoint, {cameraID, isRealFire, timestamp}).then(() => {
+              if (isUndo) {
+                delete f.voted
+              } else {
+                f.voted = isRealFire
+              }
+
+              updateFires(indexOfOldFires > -1)
```

Comment:
Should this be done once per API call or once at the end after all APIs calls are issued?

Reply:
Good catch. I missed this while copy-pasting from single-vote to looping.